packages/test/test-end-to-end-tests/src/test/offline/offlineTestsUtils.ts

```diff
@@ -0,0 +1,131 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { strict as assert } from "assert";
+
+import type { IContainer } from "@fluidframework/container-definitions/internal";
+import type { IContainerExperimental } from "@fluidframework/container-loader/internal";
+import type { IRequest } from "@fluidframework/core-interfaces";
+import { Deferred } from "@fluidframework/core-utils/internal";
+import type { IDocumentServiceFactory } from "@fluidframework/driver-definitions/internal";
+import { toDeltaManagerInternal } from "@fluidframework/runtime-utils/internal";
+import {
+	type ITestFluidObject,
+	type ITestContainerConfig,
+	type ITestObjectProvider,
+	waitForContainerConnection,
+	toIDeltaManagerFull,
+} from "@fluidframework/test-utils/internal";
+
+import { wrapObjectAndOverride } from "../../mocking.js";
+
+type SharedObjCallback = (
+	container: IContainer,
+	dataStore: ITestFluidObject,
+) => void | Promise<void>;
+
+/**
+ * load container, pause, create (local) ops from callback, then optionally send ops before closing container
+ */
+export const getPendingOps = async (
```

Comment:
Maybe call this `loadContainerAndGetPendingState`. It's confusing that `getPendingOps` would load a container as well.

Reply:
Or maybe `generatePendingState`?  It needs a live container to do so, but it closes that container before returning, so I wouldn't put `loadContainer` in the title.